ethereum-network-partition-test/main_test.go

```diff
@@ -358,112 +462,122 @@ func waitUntilAllNodesGetSyncedBeforeInducingThePartition(
+				}
+				if block.NumberU64() == blockNumberToWaitForOnEachNode {
+					ethNodeBlocksByServiceId.Store(node0ServiceId, block)
+					logrus.Infof("Node '%s' produced one block after partitioning (block number '%d', block hash '%s')", node0ServiceId, block.NumberU64(), block.Hash().Hex())
 				}
-				ethNodeBlocksByServiceId.Store(node0Id, block)
-
 			}()
 
-			//node2
+			//nodeN
 			go func() {
 				defer wg.Done()
-				block, err := getMostRecentNodeBlockWithRetries(ctx, node2Id, node2Client, retriesAttempts, retriesSleepDuration)
+				logrus.Debugf("Checking block number for node '%s'", nodeNServiceId)
+				if nodeNReachedExpectedBlockNumber {
+					logrus.Debugf("Node '%s' already reached block number '%d'", nodeNServiceId, blockNumberToWaitForOnEachNode)
+					return
+				}
+				block, err := getMostRecentNodeBlockWithRetries(ctx, nodeNServiceId, node2Client, retriesAttempts, retriesSleepDuration)
 				if err != nil {
-					errorChanForCheckingNode0 <- stacktrace.Propagate(err, "An error occurred getting the mos recent node bloc for service '%v'", node2Id)
+					errorChan <- stacktrace.Propagate(err, "An error occurred getting the mos recent node bloc for service '%v'", nodeNServiceId)
+				}
+				if block.NumberU64() == blockNumberToWaitForOnEachNode {
+					ethNodeBlocksByServiceId.Store(nodeNServiceId, block)
+					logrus.Infof("Node '%s' produced one block after partitioning (block number '%d', block hash '%s')", nodeNServiceId, block.NumberU64(), block.Hash().Hex())
 				}
-				ethNodeBlocksByServiceId.Store(node2Id, block)
 			}()
-
 			wg.Wait()
-
-			uncastedNode0Block, ok := ethNodeBlocksByServiceId.LoadAndDelete(node0Id)
-			if !ok {
-				return 0, "", stacktrace.NewError("An error occurred loading the node's block for service with ID '%v', the value for ley '%v' was no loaded", node0Id, node0Id)
+			_, node0ReachedExpectedBlockNumber = ethNodeBlocksByServiceId.Load(node0ServiceId)
+			_, nodeNReachedExpectedBlockNumber = ethNodeBlocksByServiceId.Load(nodeNServiceId)
+		case err := <-errorChan:
+			if err != nil {
+				return 0, "", 0, "", stacktrace.Propagate(err, "An error occurred checking for next partition1 block number and hash")
 			}
-			node0Block := uncastedNode0Block.(*types.Block)
+			return 0, "", 0, "", stacktrace.NewError("Something unexpected happened, a new value was received from the error channel but it's nil")
+		}
+	}
 
-			uncastedNode2Block, ok := ethNodeBlocksByServiceId.LoadAndDelete(node2Id)
-			if !ok {
-				return 0, "", stacktrace.NewError("An error occurred loading the node's block for service with ID '%v', the value for ley '%v' was no loaded", node2Id, node2Id)
-			}
-			node2Block := uncastedNode2Block.(*types.Block)
+	uncastedNode0Block, ok := ethNodeBlocksByServiceId.LoadAndDelete(node0ServiceId)
+	if !ok {
+		return 0, "", 0, "", stacktrace.NewError("An error occurred loading the node's block for service with ID '%v', the value for key '%v' was no loaded", nodeNServiceId, nodeNServiceId)
```

Comment:
s/no loaded/not loaded/

and in this case, I think `ok` should be `found`

Reply:
It is actually `loaded` in this particular case, but good catch